src/code/z_collision_check.c

```diff
@@ -1,392 +1,3345 @@
+}
+
+// Draw Collider
+void func_8005D4DC(GlobalContext* globalCtx, Collider* collider) {
+    ColliderJntSph* jntSph;
+    ColliderCylinder* cylinder;
+    ColliderTris* tris;
+    ColliderTrisItem* trisItem;
+    ColliderQuad* quad;
+    s32 i;
+
+    if (collider == NULL) {
+        return;
+    }
+    switch (collider->shape) {
+        case COLTYPE_JNTSPH:
+            jntSph = (ColliderJntSph*)collider;
+            for (i = 0; i < jntSph->count; i++) {
+                func_800D05D0(globalCtx, &jntSph->list[i].dim.worldSphere);
+            }
+            break;
+        case COLTYPE_CYLINDER:
+            cylinder = (ColliderCylinder*)collider;
+            func_800D05DC(globalCtx, &cylinder->dim);
+            break;
+        case COLTYPE_TRIS:
+            tris = (ColliderTris*)collider;
+            for (i = 0; i < tris->count; i++) {
+                trisItem = &tris->list[i];
+                func_8005B280(globalCtx->state.gfxCtx, &trisItem->dim.vtx[0], &trisItem->dim.vtx[1],
+                              &trisItem->dim.vtx[2]);
+            }
+            break;
+        case COLTYPE_QUAD:
+            quad = (ColliderQuad*)collider;
+            func_8005B280(globalCtx->state.gfxCtx, &quad->dim.quad[2], &quad->dim.quad[3], &quad->dim.quad[1]);
+            func_8005B280(globalCtx->state.gfxCtx, &quad->dim.quad[1], &quad->dim.quad[0], &quad->dim.quad[2]);
+            break;
+    }
+}
+
+void CollisionCheck_Draw(GlobalContext* globalCtx, CollisionCheckContext* check) {
+    Collider* collider;
+    s32 i;
+
+    if (AREG(15)) {
+        if (AREG(21)) {
+            for (i = 0; i < check->colAtCount; i++) {
+                func_8005D4DC(globalCtx, check->colAt[i]);
+            }
+        }
+        if (AREG(22)) {
+            for (i = 0; i < check->colAcCount; i++) {
+                func_8005D4DC(globalCtx, check->colAc[i]);
+            }
+        }
+        if (AREG(23)) {
+            for (i = 0; i < check->colOcCount; i++) {
+                collider = check->colOc[i];
+                if (collider->maskA & 1) {
+                    func_8005D4DC(globalCtx, collider);
+                }
+            }
+        }
+        if (AREG(24)) {
+            func_80042C3C(globalCtx, &globalCtx->colCtx);
+        }
+        if (AREG(25)) {
+            func_80042FC4(globalCtx, &globalCtx->colCtx);
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_collision_check/func_8005D3BC.s")
+s32 (*D_8011DEF8[])(GlobalContext*, Collider*) = { Collider_JntSphSetAT, Collider_CylinderSetAT, Collider_TrisSetAT,
+                                                   Collider_QuadSetAT };
+
+s32 CollisionCheck_SetAT(GlobalContext* globalCtx, CollisionCheckContext* check, Collider* collider) {
+    s32 index;
+
+    if (func_800C0D28(globalCtx) == 1) {
+        return -1;
+    }
+    if (collider->shape >= 4) {
+        __assert("pcl_obj->data_type <= CL_DATA_LBL_SWRD", "../z_collision_check.c", 2997);
+    }
+    check = (void*)check;
```

Comment:
What's the purpose of this line?

Reply:
thought it was necessary for ok. removing it is fine.